test_programs/compile_success_empty/regression_4635/src/main.nr

```diff
@@ -42,8 +42,8 @@ struct MyStruct<T> {
     a: T
 }
 
-impl<T> Deserialize<1> for MyStruct<T> {
-    fn deserialize(fields: [Field; 1]) -> Self where T: FromField {
+impl<T> Deserialize<1> for MyStruct<T> where T: FromField {
```

Comment:
This had to be updated as this PR now triggers an error for impls stricter than their traits.

Reply:
Yeah, we shouldn't have allowed this to begin with